### ISSUE TITLE
Allow `match-document-filename` to handle abbreviations

### ISSUE THREAD

**Issue by CraigSiemens** [dotansimha/graphql-eslint#930]
**Is your feature request related to a problem? Please describe.**

Say I have a query for tv series that are shown on the channel ABC. I'd make the query as 
```GraphQL
query ABCSeries {
   ...
}
```

When the `match-document-filename` rule runs with `query: { style: PascalCase }`, it expects the file name to be `AbcSeries` which isn't want we'd like.

**Describe the solution you'd like**

We'd like the filename to be enforced as `ABCSeries`. I'd be good if there was a way to allow abbreviations to be kept in the same case in the file name. Some possible ways I thought of doing it could be:
- Update `PascalCase` to preserve the capitalization from the query/mutation/fragment name.
- Add another style `same` that just enforces that the file name is the same as the type name.
- Add an option to define abbreviations. Then `PascalCase` and `camelCase` could use that to decide whether it should be all lowercase (when the first word in `camelCalse`) or uppercase.

**Comment by B2o5T** [comment]
duplicate of https://github.com/dotansimha/graphql-eslint/issues/919
new option `matchDocumentStyle` will be added in next version of `graphql-eslint`
